### PR TITLE
Improve resilience to COM RCW cleanup issues in tests

### PR DESCRIPTION
**Pull request by sharwell**
Analysis of heap dumps in https://github.com/dotnet/roslyn/issues/49024 suggests that the test process is attempting to clean up COM RCWs after the STA thread is no longer available. Local testing suggests a strategy of aborting the STA thread during an AppDomain unload request mitigates the delays.

### CODE REVIEW

File: src/EditorFeatures/TestUtilities/Threading/StaTaskScheduler.cs [hunk 1]
```diff
@@ -45,8 +46,7 @@ static StaTaskScheduler()
             // to DomainUnload is a reasonable place to do it.
             AppDomain.CurrentDomain.DomainUnload += (sender, e) =>
             {
-                GC.Collect();
-                GC.WaitForPendingFinalizers();
+                GC.GetTotalMemory(forceFullCollection: true);
```
Comment (jasonmalinowski) [inline src/EditorFeatures/TestUtilities/Threading/StaTaskScheduler.cs:49]: Why this change?

**Review by jasonmalinowski — CHANGES_REQUESTED** [review]
@sharwell Can this pull request be updated to describe the problems that we are trying to improve? Otherwise it's unclear what the goal is.

**Comment by sharwell** [comment]
Marking as draft since it didn't fix the integration test case
